org.eclipse.search/search/org/eclipse/search/internal/core/text/TextSearchVisitor.java

```diff
@@ -223,7 +223,10 @@ public IStatus processFile(List<IFile> sameFiles, IProgressMonitor monitor) {
 						}
 						occurences = locateMatches(file, charsequence, matcher, monitor);
 					} catch (FileCharSequenceProvider.FileCharSequenceException e) {
-						throw (RuntimeException) e.getCause();
+						if (e.getCause() instanceof RuntimeException) {
+							throw (RuntimeException) e.getCause();
+						}
```

Comment:
Let's try Java 17 way:
```suggestion
						if (e.getCause() instanceof RuntimeException runtimeEx) {
							throw runtimeEx.getCause();
						}
```

Reply:
Weirdly, when I change to the Java 17 format, I get an error "Unhandled exception type Throwable".  Do you see this too?  With the previous version, no error.

Reply:
You don't want to call getCause on runtimeEx.  It's already thing you want to throw.  😬